packages/react-widgets/src/hooks/useWidgetFetch.js

```diff
@@ -57,6 +57,19 @@ export function selectGeometryToIntersect(global, viewport, spatialFilter) {
   }
 }
 
+// stolen from deck.gl/modules/carto/src/layers/h3-tileset-2d.ts
+const BIAS = 2;
+export function getHexagonResolution(viewport, tileSize) {
+  // Difference in given tile size compared to deck's internal 512px tile size,
+  // expressed as an offset to the viewport zoom.
+  const zoomOffset = Math.log2(tileSize / 512);
+  const hexagonScaleFactor = (2 / 3) * (viewport.zoom - zoomOffset);
+  const latitudeScaleFactor = Math.log(1 / Math.cos((Math.PI * viewport.latitude) / 180));
+
+  // Clip and bias
+  return Math.max(0, Math.floor(hexagonScaleFactor + latitudeScaleFactor - BIAS));
+}
+
```

Comment:
(no action required) @felixpalmer maybe this is something we'll want to pull into `@carto/api-client` to reuse in both places? Once the repository is public and can be made a dependency, that is.

Reply:
Hmm, wouldn't it be better to export it from deck? It's a key part of how `H3Tileset` works

Reply:
Oh, yes – that does sound better. :)